include/dma.h

```diff
@@ -128,9 +128,14 @@ struct dma_block_config {
  *
  *     callback_arg  private argument from DMA client.
  *
- * dma_callback is the callback function pointer. If enabled, callback function
- *              will be invoked at transfer completion or when error happens
- *              (error_code: zero-transfer success, non zero-error happens).
+ * dma_ht_callback is the callback function pointer of half transfer complete
```

Comment:
Candid question:
Why adding a new callback while we could use the same callback and check the Event Flag to know which event triggered the interrupt?
Also you might need to extend/document usage of complete_callback_en that should be used to configure this behavior

Reply:
Additional question: there are other modes of DMA (I'm thinking to configurable thresholds in FIFO mode) that would require to be supported as well in the future and then I don't think this approach of adding a callback per mode would fit..

Reply:
Using a single callback (e.g. dma_callback) and checking TC or HT inside it may also work I guess. Maybe it was done like that for simplicity?

Reply:
I also think that adding a new callback is not a good approach. I just found that 'callback_arg' argument in 'dma_config' can do this job. Client like i2s driver of stm32 passes a dma driver defined struct pointer to the driver, and when driver calls it's client's irq handler, client may check some enum variable in this struct to know which interrupt that is. In this way, adding new modes are also much easier and we can only apply changes to the dma drivers and clients of stm32.